### ISSUE TITLE
Support label-channels (particularly important for cuda)

### ISSUE THREAD

**Issue by pablovela5620** [prefix-dev/pixi#567]
### Problem description

I often need to change different versions of cuda, for example my machine has 12.3 installed, while I need to compile things like [diff-gaussian-rasterization-w-depth](https://github.com/JonathonLuiten/diff-gaussian-rasterization-w-depth/tree/cb65e4b86bc3bd8ed42174b72a62e8d3a3a71110 ) with cuda 11.6 or [tiny-cuda-nn](https://github.com/NVlabs/tiny-cuda-nn) with 11.8

as it stands I can't install different versions of cuda via pixi. Using the nvidia channel and selecting the version of cuda always defaults to 12.3

**Comment by ruben-arts** [comment]
He,

We support channel specific dependencies. This means you can define from which channel you want your [dependencies](https://pixi.sh/configuration/#the-dependencies-tables) to come. 

In this case I tested the following project and that seems to work:
```toml
[project]
name = "cuda"
channels = ["conda-forge", "nvidia/label/cuda-11.8.0"]
platforms = ["linux-64"]

[dependencies]
cuda = {version = "*", channel="nvidia/label/cuda-11.8.0"}
```

Hope this helps

**Comment by pablovela5620** [comment]
Thanks Ruben! I tried running that exact pixi.toml but find the following problem
running this
```
pablo@pablo-ubuntu:~/0Dev/repos/ns-test$ pixi run nvcc --version
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2022 NVIDIA Corporation
Built on Mon_Oct_24_19:12:58_PDT_2022
Cuda compilation tools, release 12.0, V12.0.76
Build cuda_12.0.r12.0/compiler.31968024_0
pablo@pablo-ubuntu:~/0Dev/repos/ns-test$ pixi run which nvcc
/home/pablo/0Dev/repos/ns-test/.pixi/env/bin/nvcc
pablo@pablo-ubuntu:~/0Dev/repos/ns-test$ 
```

for some reason cuda 12.0 is being installed instead of 11.8

If I run the following through mamba

```
pablo@pablo-ubuntu:~/0Dev/repos/ns-test$ mamba env create -n cuda

pablo@pablo-ubuntu:~/0Dev/repos/ns-test$ mamba activate cuda

(cuda) pablo@pablo-ubuntu:~/0Dev/repos/ns-test$ which nvcc
/home/pablo/micromamba/envs/cuda/bin/nvcc
(cuda) pablo@pablo-ubuntu:~/0Dev/repos/ns-test$ nvcc --version
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2022 NVIDIA Corporation
Built on Wed_Sep_21_10:33:58_PDT_2022
Cuda compilation tools, release 11.8, V11.8.89
Build cuda_11.8.r11.8/compiler.31833905_0
(cuda) pablo@pablo-ubuntu:~/0Dev/repos/ns-test$ 
```
This results in 11.8 being installed just fine

**Comment by ruben-arts** [comment]
Ah yeah you are right. If you want all your cuda package from the nvidia label channel you should make it the first in the list of channels. 

You are only getting the packages not on conda-forge from the label channel with my previous config. 

```toml
channels = ["nvidia/label/cuda-11.8.0", "conda-forge"]
``` 
This should do the trick. 

Now all packages that are available on the `nvidia/label/cuda-11.8.0` channel will be fetch from that channel. And only the packages not available there will be pulled from `conda-forge` this is called `strict channel priority` which is our default behavior.
